server/controllers/clubProfileController.js

```diff
@@ -6,15 +9,30 @@ module.exports = {
     },
     getById: function(req, res) {
         // TODO; req.params.id
+        clubProfile.findById(req.params.id)
+                .then(clubprofile => res.json(clubprofile))
+                .catch(err => res.status(422).json(err));
+
     },
     create: function(req, res) {
         // TODO; req.body contains profile information
+        clubProfile.create(req.body)
+                .then(newclubProfile => res.json(newclubProfile))
+                .catch(err => res.status(422).json(err))
     },
     update: function(req, res) {
         // TODO; req.body contains profile information and req.params.id
+        clubProfile.findOneAndUpdate({ _id: req.params.id}, req.body)
+                .then(clubprofile => res.json(clubprofile))
+                .catch(err => res.status(422).json(err))
     },
     delete: function(req, res) {
         // TODO: req.params.id
+        clubProfile.findById({ _id: req.params.id })
```

Comment:
what's the reason for using `findById` differently here vs. earlier in the `getById` function?

Reply:
They are the same, get by Id would be used by the frontend. getById will have different usage in the future as discussed in out meeting